src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/TaskReplicator.cs

```diff
@@ -131,17 +132,27 @@ private TaskReplicator(ParallelOptions options, bool stopOnFirstFailure)
 
         public static void Run<TState>(ReplicatableUserAction<TState> action, ParallelOptions options, bool stopOnFirstFailure)
         {
-            int maxConcurrencyLevel = (options.EffectiveMaxConcurrencyLevel > 0) ? options.EffectiveMaxConcurrencyLevel : int.MaxValue;
-
-            TaskReplicator replicator = new TaskReplicator(options, stopOnFirstFailure);
-            new Replica<TState>(replicator, maxConcurrencyLevel, CooperativeMultitaskingTaskTimeout_RootTask, action).Start();
-
-            Replica? nextReplica;
-            while (replicator._pendingReplicas.TryDequeue(out nextReplica))
-                nextReplica.Wait();
-
-            if (replicator._exceptions != null)
-                throw new AggregateException(replicator._exceptions);
+#pragma warning disable CS0162
+            if (System.Threading.Tasks.Parallel.IsSingleThreadedHost) {
+                var timeout = GenerateCooperativeMultitaskingTaskTimeout();
```

Comment:
No.  This is a self-imposed timeout to enable work items to recycle themselves to avoid hogging thread pool threads.  A task starts running, and after X amount of time, rather than continuing to process the loop in this task, it queues a replica of itself and exits, in order to give the thread pool the option of retiring this thread, running other stuff on it, etc.